components/application-connectivity-validator/internal/validationproxy/handler.go

```diff
@@ -82,6 +101,34 @@ func (ph *proxyHandler) ProxyAppConnectorRequests(w http.ResponseWriter, r *http
 	reverseProxy.ServeHTTP(w, r)
 }
 
+func (ph *proxyHandler) getCompassMetadataClientIDs(applicationName string) ([]string, apperrors.AppError) {
+	applicationClientIDs, found := ph.cache.GetClientIDs(applicationName)
+	if !found {
+		var err apperrors.AppError
+		applicationClientIDs, found, err = ph.getApplicationClientIDs(applicationName)
+		if err != nil {
+			return []string{}, err
+		}
+
+		if found {
+			ph.cache.SetClientIDs(applicationName, applicationClientIDs)
+		}
```

Comment:
Correct me if I am wrong, but from what I understand if `CompassMetadata` is `nil` we do not save anything in cache. As a result, for standalone Applications, we do the API Server call every time which is not desired behavior. We should either cache the empty array for this app or skip lookup all together.

Reply:
Done!